src/eth_provider/provider.rs

```diff
@@ -552,27 +552,11 @@ where
         let filter = into_filter("tx.hash", &transaction_signed.hash, HASH_HEX_STRING_LEN);
         let pending_transaction = self.database.get_one::<StoredPendingTransaction>(filter.clone(), None).await?;
 
+        // Number of retries for the transaction (0 if it's a new transaction)
+        let retries = pending_transaction.as_ref().map(|tx| tx.retries).unwrap_or_default();
```

Comment:
in what case can the retries be undefined?
if so, why do we want to make it default?

Reply:
`get_one` returns an Option, so if it's Some, we take the current retries, otherwise the current retries are 0 (default for u8).

Reply:
but i'm saying, if its none then the tx is not in DB?

Reply:
yes yes, but it's the pending_transaction db, so if it's not there, it means it's just never been retried.